### DISCUSSION TITLE
NPC Overhauls + Botox For Skyrim

### DISCUSSION THREAD

**Discussion by MadKillerChicken** [focustense/easymod#183] [Q&A] [answered]
Recently found this great project and am giving it a try.
It's insanely simple to use and works great, until I mix in [`Better Vampires`](https://www.nexusmods.com/skyrimspecialedition/mods/1925) and [`Better Vampire NPCs NAC*`](https://www.nexusmods.com/skyrimspecialedition/mods/9510).
(*_The BV NAC version shouldn't apply any visual changes_)

Before using EasyNPC I had these vampire mods, `Botox For Skyrim` and `Bijin AIO` in my load order.
It was working as expected, visually at least, everyone, even the vampires, was properly replaced by the visual overhauls.

Once I started using EasyNPC, to include a few other visual NPC overhauls, it worked flawlessly as before, except for the vampires.
EasyNPC reports that the vampire templates (`EnvVampireXXTemplate`) are conflicting overrides.
I tried changing load order switching different visual overhauls before or after the vampire mods, nothing seems to solve this.
The game works, no crashes, no black faces, but they don't get replaced as they did before.

![image](https://user-images.githubusercontent.com/19254972/235678784-3479f02f-0f42-4cb3-87f5-5afb5d019637.png)

I might be wrong, as I'm not entirely sure how all of this works, but I'm starting to think that load order isn't the issue.
`Better Vampires` (_not `Better Vampire NPCs NAC`_) might be changing appearances **and** acting as a master file for those vampires. Meaning I would need to disable those vampire mods after a merge, resulting in good looking vampires but missing the gameplay changes provided by those mods.

Any help would be massively appreciated, and thank you very much for your hard work with the project!

**Comment by focustense** [comment]
These are Template NPCs. Does one of the mods in question make them not-templates? I can imagine some scenario in which the Default Plugin (which controls template behavior) changes a template NPC to not-template, or at least to not Inherit Traits, but the Face Plugin/Mod doesn't have a facegen because it assumes that they are a template.

(Template NPCs who inherit traits don't really use their own facegens - they sometimes seem to have facegens in the vanilla BSAs, but I'm pretty sure the game ignores them if they do.)

You're certainly correct that it has nothing whatsoever to do with load order; the fact is that these facegens don't exist _at all_, not in any mod anywhere in your load order, which is what the app is telling you here. As you can see here, they don't exist in vanilla:

![image](https://user-images.githubusercontent.com/84610172/235828833-6ccf8790-1792-4baf-b859-9b5961927cd4.png)

The salient question is why the app thinks they _should_ exist. Post a screenshot of the NPC's template settings in xEdit, that might give us a clue.

**Reply by MadKillerChicken** [reply]
Thank you very much for helping me out!

If I correctly understand SSEEdit (screenshot below), it appears that something along the lines of your assumption is going on.
It appears that BV NPC is removing the template from all those vampires.
Am I right in assuming that there's no real solution to this?
Aside from removing/editing the BV NPC mod, or hoping that EasyNPC might address this issue in the future, if possible.


![image](https://user-images.githubusercontent.com/19254972/235878181-6d7f231f-7bd5-41ac-b2b9-82caf66e11b9.png)

**Reply by MadKillerChicken** [reply]
So, out of desperation (gotta love the drama😄) I tried the BV NPC variant that *changes* vampire appearance,
since I was using the one that *doesn't*.

Well, the vampires changed, now they look like the ones out of Buffy the Vampire Slayer (yes, I'm that old).
However, they still have the same old faces, so it didn't work, no surprise there.
The only difference is that EasyNPC now knows which mod is to blame, and instructs me to disable it.
Which I obviously can't do because it's a master and I actually wanted to use it, but I repeat myself...

I also was unable to see any differences in SSEEdit, between this and the previous version of the mod,
all template related records look the same.

**Comment by MadKillerChicken** [comment, marked as answer]
So, after some trial and error I've got a **solution of sorts**.
The issue appears to be some sort of incompatibility between Botox and EasyNPC.

1. Botox has an ESP, but it's empty and only used to load the BSA (containing FaceGen data)
2. Thus, I suppose, EasyNPC is unable to identify Botox as an appearance overhaul

If I correctly understand the "mugshot" feature, the mods there are sorted by priority from left to right,
the right-most mod with face-data is the one that gets auto-selected to provide the face.
In all cases I checked, where Botox was providing face-data, the mod was listed with the highest priority to the right.
However it was flagged as "not loaded" and instead the next best face-data mod was selected to provide the face.

In cases where no other appearance overhaul was available, where Botox should've worked as "base-line", EasyNPC would select skyrim.esm. And by doing so "enforcing" the vanilla face, as shows in this screenshot.
![Screenshot 2023-05-05 131623 (Small)](https://user-images.githubusercontent.com/19254972/236449362-c0d031fa-1557-4f14-9e6c-5f4b9b0fcaa5.jpg)

This is what happens with those vampire NPCs, but also all those NPCs that aren't covered by any overhaul.
It's however possible to **avoid this behavior, by manually selecting Botox for every NPC that was "reverted" to skyrim.esm**.

I'm not sure if EasyNPC can or even should be adapted to include mods such as Botox.
However, one thing that's probably worth noting, is that no matter how many mods provide face-data for a given NPC,
Botox is always listed as the last mod. Possibly implying that, if EasyNPC could make use of it, it would override all other mods.
![Screenshot 2023-05-05 131654 (Small)](https://user-images.githubusercontent.com/19254972/236449660-29f456f7-6a86-4f1b-90ec-5309d75097c0.jpg)

**Reply by MadKillerChicken** [reply]
I'm marking this as answer, as I'm unsure if this can/should or will be fixed.
It's a better than nothing solution.
